src/core/Form/Select/BaseSelect/SelectItemList/SelectItemList.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../../../theme';
+import { HtmlDivWithRef, HtmlUlWithRef } from '../../../../../reset';
+import { SelectItemProps } from '../SelectItem/SelectItem';
+import { baseStyles } from './SelectItemList.baseStyles';
+
+const baseClassName = 'fi-select-item-list';
+
+const selectItemListClassNames = {
+  content_wrapper: `${baseClassName}_content_wrapper`,
+};
+
+export interface SelectItemListProps {
+  /** SelectItemList container div class name for custom styling. */
+  className?: string;
+  children:
+    | Array<React.ReactElement<SelectItemProps>>
+    | React.ReactElement<SelectItemProps>;
+  forwardRef: React.RefObject<HTMLUListElement>;
+  focusedDescendantId: { id: string };
+  onBlur?: (event: React.FocusEvent<Element>) => void;
+  /** Id needed for aria-owns and aria-controls */
+  id: string;
+}
+
+class BaseSelectItemList extends Component<
+  SelectItemListProps & SuomifiThemeProp
+> {
+  private wrapperRef: React.RefObject<HTMLDivElement>;
+
+  constructor(props: SelectItemListProps & SuomifiThemeProp) {
+    super(props);
+    this.wrapperRef = React.createRef();
+  }
+
+  componentDidUpdate(prevProps: SelectItemListProps & SuomifiThemeProp) {
+    if (
+      this.props.focusedDescendantId !== prevProps.focusedDescendantId &&
+      !!this.props.focusedDescendantId.id
+    ) {
+      this.scrollItemList(this.props.focusedDescendantId.id);
+    }
+  }
+
+  private scrollItemList = (elementId: string) => {
+    if (this.wrapperRef !== null && this.wrapperRef.current !== null) {
+      const elementOffsetTop =
+        document.getElementById(elementId)?.offsetTop || 0;
+      const elementOffsetHeight =
+        document.getElementById(elementId)?.offsetHeight || 0;
+      if (elementOffsetTop < this.wrapperRef.current.scrollTop) {
+        // 4px reduction to scroll position is required due to container padding.
+        this.wrapperRef.current.scrollTop = elementOffsetTop - 4;
+      } else {
+        const offsetBottom = elementOffsetTop + elementOffsetHeight;
+        const scrollBottom =
+          this.wrapperRef.current.scrollTop +
+          this.wrapperRef.current.offsetHeight;
+        if (offsetBottom > scrollBottom) {
+          // 4px reduction to scroll position is required due to container padding.
+          this.wrapperRef.current.scrollTop =
+            offsetBottom - this.wrapperRef.current.offsetHeight - 4;
```

Comment:
Would it make sense to have the common variable for the reduction and having the comment only there (not needed if variable tells it already).
e.g
```tsx
// 4px reduction to scroll position is required due to container padding.
const scrollPositionFix = 4;
```